### PR TITLE
Using new ProxyCommit tuple to address proxying with optional data

### PR DESCRIPTION
**Pull request by jwalapr**
- PR still needs to update tests
- Needs a second pass over proxy logic
- Consider refactoring proxy pipeline

### CODE REVIEW

**Comment by jwalapr** [comment]
resolve #41

**Comment by jwalapr** [comment]
@nonameplum I did a quick pass with using a tuple to handle the new data, it changes the API a bit for calling next() so I wanted to give you a heads up. I'll revisit this tomorrow.

File: Source/Routing.swift [hunk 1]
```diff
@@ -214,6 +212,11 @@ public final class Routing: RouteOwner {
             return
         }
 
+        var data = data
+        if let newData = proxyCommit?.data {
+            data = newData
```
Comment (jwalapr) [inline Source/Routing.swift]: needed this to pass newData below to line 222.

**Comment by jwalapr** [comment]
worth noting passing (nil, nil, nil) to next will stop the route from completing.

File: Source/Routing.swift [hunk 2]
```diff
@@ -173,34 +173,32 @@ public final class Routing: RouteOwner {
                          matching route: Route,
```
Comment (nonameplum) [inline Source/Routing.swift]: I don't know why compiler doesn't see the problem in `open` method when it class process with optional data where `proxy` don't support optional. Nevertheless data parameter should be optional `data: Data?`

**Comment by nonameplum** [comment]
I don't know why compiler doesn't see the problem in open method when it class process with optional data where proxy don't support optional. Nevertheless data parameter of the `process` method should be optional `data: Data?`

**Pull request merged**